static/js/services/http-wrapper.js

```diff
@@ -32,13 +37,13 @@ var _ = require('underscore');
     '$http', '$q', 'ActiveRequests',
     function($http, $q, ActiveRequests) {
 
-      var wrap = function(args, fn) {
+      var wrap = function(args, fn, updateTarget) {
```

Comment:
Makes sense, although I've not seen the `messages` route used - it's always `messages.details`.  So I think we'd have to manually apply the "route" name to each request, rather than deriving it automatically.  Conflating 'the route the request was triggered from' with 'the route that this request nominally updates' seems confusing.  Perhaps it's incomplete understanding on my part though.